app/controllers/concerns/verify_sp_attributes_concern.rb

```diff
@@ -0,0 +1,49 @@
+module VerifySPAttributesConcern
+  def needs_verify_sp_attributes?
+    sp_session_identity.nil? || !requested_attributes_verified?
+  end
+
+  def verify_sp_attributes_url
+    sign_up_completed_url
+  end
+
+  def update_verified_attributes
+    IdentityLinker.new(
```

Comment:
Why do we need to use the `IdentityLinker` here? Re-linking the identity to update attributes means we get all of the side effects that come along with linking the identity. What are you thoughts on a new service for updating the verified the attributes?

Or, since you have `sp_session_identity`, you could just call `update!` with the merged attributes, right?

Reply:
So the issue with this is we are calling this method from `completions_controller#update`. When creating a new account, this screen is shown before an identity is created.

We could add a conditional and only call linker with `sp_session_identity` is nil?